src/main/java/software/amazon/ionpathextraction/PathExtractorImpl.java

```diff
@@ -145,24 +149,47 @@ private int invokeCallback(final IonReader reader,
         return stepOutTimes;
     }
 
-    private boolean pathComponentMatches(final SearchPath<T> searchPath,
-                                         final IonReader reader,
-                                         final int currentDepth,
-                                         final int currentPosition) {
-        List<PathComponent> pathComponents = searchPath.getPathComponents();
-
-        // currentDepth 0 can only match the empty search path: ()
-        if (currentDepth == 0) {
-            return pathComponents.isEmpty();
-        } else if (currentDepth <= pathComponents.size()) {
-            return pathComponents.get(currentDepth - 1).matches(reader, currentPosition, config);
+    private boolean searchPathPartialMatchAt(final SearchPath<T> searchPath,
+                                             final IonReader reader,
+                                             final int currentDepth,
+                                             final int currentPosition) {
+        switch (searchPath.getType()) {
+            case TOP_LEVEL:
+                return currentDepth == 0;
+
+            case ANNOTATED_TOP_LEVEL:
+                AnnotatedTopLevelSearchPath<T> annotatedSearchPath = (AnnotatedTopLevelSearchPath<T>) searchPath;
+                return currentDepth == 0
+                    && arrayEquals(
+                        reader.getTypeAnnotations(),
```

Comment:
Do we have a plan for allowing subset annotation matches? I can imagine wanting to match anything with annotation `B`, including value `A::B::C::21`.

Reply:
Not yet. We are trying to avoid adding more functionality to the path extractors without a user request. 

Annotation support is being added so we can add a path extraction alternative to the existing sparse read examples in the cookbook